src/components/MLTaskManager/ProcessAndOutput.js

```diff
@@ -51,6 +61,29 @@ class ProcessAndOutput extends React.Component {
     
     new Response(csvData).body.pipeTo(fileStream).then(onSuccess, onError)
   }
+
+  doRetire () {
+    const workflowOutput = this.context.workflowOutput
+    const mlSelection = this.context.mlSelection
+    const selection = mlSelection.selection.toJSON() || []
+    const subjectIds = getUniqueSubjectIds(selection)
+    
+    workflowOutput.retire(subjectIds)
+  }
+}
+
+function getUniqueSubjectIds (selection) {
+  if (!selection) return []
+  return selection
+  .map(image => image.meta && image.meta.subject_id)
+  .sort()
+  .map((subject_id, index, array) => {
+    if (index <= 0) return subject_id;
+    const prev = array[index - 1]
+    if (subject_id === prev) return undefined
+    return subject_id
+  })
+  .filter(subject_id => subject_id !== undefined)
 }
```

Comment:
You can remove duplicates from a JavaScript array with something like
```js
uniqueSubjectIDs = subjectIDs.filter((subjectID, index) => subjectIDs.indexOf(subjectID) === index);
```
where the filter callback only returns true if this is the first instance of `subjectID` in `subjectIDs`.

Reply:
That is _way_ better, thanks! 👍